nilearn/decoding/tests/test_decoder.py

```diff
@@ -308,6 +361,66 @@ def test_parallel_fit(rand_X_Y):
             assert a == b
 
 
+@pytest.mark.parametrize(
+    "estimator, param_name, fitted_param_name, is_classification",
+    [
+        (RidgeCV(), "alphas", "best_alpha", False),
+        (RidgeClassifierCV(), "alphas", "best_alpha", True),
+    ],
+)
+def test_parallel_fit_builtin_cv(
+    rand_X_Y, estimator, param_name, fitted_param_name, is_classification
+):
+    """Check that the `fitted_param_name` output of _parallel_fit is a single
+    value even if param_grid is wrapped in a list for models with built-in CV.
+    """
+    param_values = [0.001, 0.01, 0.1, 1, 10, 100, 1000]
+
+    # y will be replaced if this is a classification
+    X, y = make_regression(
+        n_samples=N_SAMPLES,
+        n_features=20,
+        n_informative=5,
+        noise=0.2,
+        random_state=42,
+    )
+
+    # train/test indices
+    n_samples_train = int(0.8 * N_SAMPLES)
+    train = range(n_samples_train)
+    test = range(n_samples_train, N_SAMPLES)
+
+    # define a screening selector
+    selector = check_feature_screening(
+        screening_percentile=None, mask_img=None, is_classification=False
+    )
+
+    # create appropriate scorer and update y for classification
+    if is_classification:
+        scorer = check_scoring(estimator, "accuracy")
+        _, y = rand_X_Y
+    else:
+        scorer = check_scoring(estimator, "r2")
+
+    for params in [param_values, [param_values]]:  # test unwrapped and wrapped
```

Comment:
this too can probably be parametrized

Reply:
not sure the indentation will be quite correct on some of the code I changed, so double checks things locally if you accept the suggestions